### PR TITLE
Add unitig-caller

### PR DESCRIPTION
**Pull request by johnlees**
:information_source:
If this is your first pull request to Bioconda please ping `@bioconda/core` so it can be reviewed and to request being added as a member of the Bioconda team. Members of the Bioconda team are able to merge their own pull requests once tests and linting have passed.

----------------

* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [x] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

Most of the work is done in the `setup.py` provided with the package

### CODE REVIEW

**Comment by johnlees** [comment]
Request for review please @bioconda/core (is this still the correct group to request review from?)

**Comment by johnlees** [comment]
Q: This worked ok, but I had intended to include `pthread-stubs` in the host section. When trying to find it, it seems to come from having the compiler tools in the build section:
```
Needed DSO x86_64-conda_cos6-linux-gnu/sysroot/lib/libpthread.so.0 found in CDT/compiler package conda-forge::gcc_impl_linux-64-7.3.0-habb00fd_1
```
Is that ok, or should I add `pthread-stubs` to the host section too?

File: recipes/unitig-caller/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+  host:
+    - python >=3
+    - pip
+    - cmake
+    - pybind11
```
Comment (dpryan79) [inline recipes/unitig-caller/meta.yaml]: Can you try moving `cmake` to the `build:` section? `pybind11` only contains headers, correct?

File: recipes/unitig-caller/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  run:
+    - python >=3
+    - pthread-stubs
```
Comment (dpryan79) [inline recipes/unitig-caller/meta.yaml]: Is this actually linking against a library from Xorg?

File: recipes/unitig-caller/meta.yaml (hunk 1 above)
Comment (johnlees) [reply recipes/unitig-caller/meta.yaml]: Yes, I believe `pybind11` is header only (so not in run). I will move `cmake`

File: recipes/unitig-caller/meta.yaml (hunk 2 above)
Comment (johnlees) [reply recipes/unitig-caller/meta.yaml]: I'm not sure exactly what you mean here, but that wasn't my intention anyway

**Comment by johnlees** [comment]
@dpryan79 Do these changes look like what you were after? I just got rid of the pthreads stuff
Did you want me to do anything with `pybind11`?

**Comment by dpryan79** [comment]
:+1: from me. No need to do anything with pybind11, I was just double checking.

**Comment by epruesse** [comment]
```
 - python >3
```
doesn't do what you want unless you also have `noarch: python` set.

Since you are linking to python, you needto build for all python versions (3.6 and 3.7), so we need to use skip and add python.

It's funky conda stuff. Should work now.

**Comment by johnlees** [comment]
Great, thank you both very much 😀

**Pull request merged**